packages/neutron-std/src/serde/mod.rs

```diff
@@ -14,7 +14,19 @@ use std::{
 struct StringOrNumberVisitor<T> {
     p: PhantomData<T>,
 }
-
+// The Visitor helps deserialize a number, both from its numerical JSON representation and from a string. 
+// For example, for the struct: 
+// ```rust
+// struct Foo {
+//     pub bar: i32;
+// }
+// ```
+// Both JSON representations `'{"foo":"12"}'` and `'{"foo":12}'` will give the same result upon deserialization:
+// ```json
+// {
+//     "foo": 12
```

Comment:
the field in JSONs must be `bar`, right?

Reply:
and why is the result of deserialization is another JSON? that's a bit confusing. I think you can just remove the result JSON and just leave the line:
```
// Both JSON representations `'{"foo":"12"}'` and `'{"foo":12}'` will give the same result upon deserialization.
```